src/client/components/pages/parts/add-to-collection-modal.js

```diff
@@ -0,0 +1,296 @@
+import React from 'react';
+import ReactSelect from 'react-select';
+import SelectWrapper from '../../input/select-wrapper';
+import _ from 'lodash';
+import request from 'superagent';
+
+
+const {Alert, Col, Button, Modal} = bootstrap;
+
+class AddToCollectionModal extends React.Component {
+	constructor(props) {
+		super(props);
+		this.state = {
+			collectionsAvailable: [],
+			message: {
+				text: null,
+				type: null
+			},
+			selectedCollections: [],
+			showCollectionForm: false
+		};
+
+		this.getCollections = this.getCollections.bind(this);
+		this.handleAddToCollection = this.handleAddToCollection.bind(this);
+		this.toggleRow = this.toggleRow.bind(this);
+		this.handleShowCollectionForm = this.handleShowCollectionForm.bind(this);
+		this.handleShowAllCollections = this.handleShowAllCollections.bind(this);
+		this.handleAddToNewCollection = this.handleAddToNewCollection.bind(this);
+		this.isValid = this.isValid.bind(this);
+	}
+
+	async componentDidMount() {
+		const collections = await this.getCollections();
+		// eslint-disable-next-line react/no-did-mount-set-state
+		this.setState({collectionsAvailable: collections});
+	}
+
+	toggleRow(collectionId) {
+		const oldSelected = this.state.selectedCollections;
+		let newSelected;
+		if (oldSelected.find(selectedId => selectedId === collectionId)) {
+			newSelected = oldSelected.filter(selectedId => selectedId !== collectionId);
+		}
+		else {
+			newSelected = [...oldSelected, collectionId];
+		}
+		this.setState({
+			selectedCollections: newSelected
+		});
+	}
+
+	async getCollections() {
+		// Get all collections of the user (unlikely that a user will have more than 10000 collections
+		const req = await request.get(`/editor/${this.props.userId}/collections/collections?type=${this.props.entityType}&size=10000`);
+		const collections = req.body;
+		return collections;
+	}
+
+	async handleAddToCollection() {
+		const {bbids} = this.props;
+		const {selectedCollections} = this.state;
+		if (selectedCollections.length) {
+			const promiseArray = [];
+			selectedCollections.forEach((collectionId) => {
+				const submissionURL = `/collection/${collectionId}/add`;
+				promiseArray.push(
+					request.post(submissionURL)
+						.send({bbids})
+				);
+			});
+			await Promise.all(promiseArray);
+			this.setState({
+				message: {
+					text: `Successfully added to selected collection${selectedCollections.length > 1 ? 's' : ''}`,
+					type: 'success'
+				}
+			});
+		}
+		else {
+			this.setState({
+				message: {
+					text: 'No Collection Selected',
+					type: 'danger'
+				}
+			});
+		}
+	}
+
+	handleShowCollectionForm() {
+		this.setState({message: {}, showCollectionForm: true});
+	}
+
+	handleShowAllCollections() {
+		this.setState({message: {}, showCollectionForm: false});
+	}
+
+	handleAddToNewCollection(evt) {
+		evt.preventDefault();
+
+		if (!this.isValid()) {
+			this.setState({
+				message: {
+					text: 'Internal Error: Try again later',
```

Comment:
I liked the "Incomplete form"  message you had previously.
How about "The form is incomplete. Please fill in a name and privacy option before continuing."